test/mpi/gentests.sh

```diff
@@ -49,9 +66,26 @@ while [ true ] ; do
             maxbufsize=$(echo $1 | cut -d= -f2)
             shift
             ;;
+        *)
+            echo Unrecognized option [$1]
+            exit 1
+            ;;
     esac
 done
 
+if test -z $maxbufsize ; then
+    probe_memory
+    if test -z $mem_available ; then
+        maxbufsize=2147483648 # default 2 GB
+    elif test $mem_available -le 3 ; then
+        maxbufsize=134217728  # 128 MB
+    elif test $mem_available -le 4 ; then
+        maxbufsize=134217728  # 512 MB
+    else
+        maxbufsize=2147483648 # default 2 GB
+    fi
+fi
+
```

Comment:
I like that this patch checks for the actual memory available on each machine.  It's more precise than the previous hack.  However, on 32-bit machines, mallocs of `>= 1GB` fail, even if the system has more memory.  So we'll need an additional check.

Reply:
Hmm.  What happens if I'm running configure on a different machine than the actual test machine?  Cross compilation is one example, but this is possible even without cross compilation (e.g., login node of a cluster).  Shouldn't we be checking for the available memory inside `MTest_Init` instead?

Reply:
To run the testsuite, we need compile each test and run, so it does really make sense to *cross-compile* testing unless significant infra-structure change, right?

Testsuite is currently configured within mpich configure. I assume we are going to separate the testsuite configure from the mpich one, i.e. configure and run testsuite independent of mpich build. My current experiment with Jenkins mpich/quick test is done that way. 

For specific case, user always can pass in `--maxbufsize=xxx` explicitly, such as the case of cluster. But even in that case, I think it makes sense to just configure and run testing on the compute node as batch job. There isn't much saving to do that on the login node anyway, is there?

We could check memory inside `MTest_Init`, or we could do that in `runtests` and implement it as a dtpool command line option. In fact, I plan to add memory checking to `runtests` anyway, then it can honor and control tests with explicit memory requirement.

Reply:
No, running "make" instead of "make testing" inside the test directory would only build the tests and not run them.  Our jenkins is an exception in that we do the build and running together.  Other partners such as Cray (and IBM for their BG/Q machine in the past) would do the build separately from the run.  So, I don't think what I'm asking for is as rare as you might think.